UnityProject/Assets/Scripts/Util/NetworkSingletonManager.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Mirror;
+using UnityEngine;
+
+public class NetworkSingletonManager<T> : NetworkBehaviour where T : NetworkBehaviour
+{
+    private static bool shuttingDown = false;
+    private static readonly object @lock = new object();
+    private static T instance;
+
+    public static T Instance
+    {
+        get
+        {
+            if (shuttingDown)
+                return null;
+
+            lock (@lock)
+            {
+                if (instance == null)
+                {
+                    instance = (T)FindObjectOfType(typeof(T));
```

Comment:
FindObjectOfType is really bad for loading times, thats why most of the managers use the awake to set the instance

Reply:
Yeah, pretty sure darkvergus used that pattern because it is what he saw in most of your singletons. Can you provide an example of the right way to do it? So we can update our old singletons in the process by simply making them inherit this base singleton class darkvergus is doing.